presto-jdbc/src/main/java/com/facebook/presto/jdbc/ObjectCasts.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Date;
+import java.sql.SQLException;
+import java.sql.Time;
+import java.sql.Timestamp;
+
+import static java.lang.Byte.parseByte;
+import static java.lang.Double.parseDouble;
+import static java.lang.Float.parseFloat;
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.lang.Short.parseShort;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+final class ObjectCasts
+{
+    private ObjectCasts() {}
+
+    public static byte castToByte(Object x, int targetSqlType)
```

Comment:
I run with a huge number of non-standard inspections enabled but this one I turned off, as it never seemed to be useful.

My philosophy is to control visibility at the class level. The class is package-private, so these methods aren't actually public. However, they are public from the perspective of the class API. We could use public or package-private interchangeably, so better to standardize on always use public, rather than being dependent on the visibility of the class.

From a practical perspective, if we wanted to make this class public in the future, all we need to do is flip the class visibility.

Reply:
I understand your arguments, but also I am able to argue the opposite ;)  I don't mind either way. I mean: i value consistent code style highly, but I am not going to argue about specific rules (well... usually). 
I think we should have clear guidelines what inspections should/must be enabled, otherwise you will end-up fixing/silencing my errors/problems, only because my IDE wasn't picky about them. (In particular, the topic of `public` method inside `private class`  is something that pops from time to time in reviews)

Reply:
Agreed, What we really need is a way to run inspections from the build. I often see commits that have issues easily caught by inspections (even default ones), so relying on everyone to use IDE inspections unfortunately isn't sufficient. But building a Maven plugin to run IntelliJ inspections would be a huge project.